plugin/completion.py

```diff
@@ -241,13 +241,29 @@ def format_completion(self, item: dict) -> 'Tuple[str, str]':
             kind = item.get("kind")
             if kind:
                 hint = completion_item_kind_names[kind]
-        # label is an alternative for insertText if insertText not provided
-        insert_text = item.get("insertText") or label
+        # label is an alternative for insertText if neither textEdit nor insertText is provided
+        insert_text = self.text_edit_text(item) or item.get("insertText") or label
         if len(insert_text) > 0 and insert_text[0] == '$':  # sublime needs leading '$' escaped.
-            insert_text = '\$' + insert_text[1:]
+            insert_text = '\\$' + insert_text[1:]
```

Comment:
I'm a bit worried about this change as it was a working fix for completions from PHP language servers. Can you tell me what is problematic, or should we take this change in a separate PR?

Reply:
It is just about escaping in Python -- `'...'` still respects backslash escape codes, `\$` is not proper escape code so Python leaves it as is, however Linters complain about this invalid escape code, hence backslash should be escaped, or raw Python string should be used.